system/ee/EllisLab/ExpressionEngine/Cli/Cli.php

```diff
@@ -0,0 +1,376 @@
+    }
+
+    /**
+     * core CLI command processing
+     * @return void
+     */
+    public function process()
+    {
+        $this->standalone = defined('EE_INSTALLED') && EE_INSTALLED == false;
+
+        $this->availableCommands = $this->availableCommands();
+
+
+        // Check if command exists
+        // If not, return
+        if (! $this->commandExists()) {
+            return $this->fail('Command not found');
+        }
+
+        $commandClass = $this->getCommand($this->commandCalled);
+
+        if (! class_exists($commandClass)) {
+            return $this->fail('Command not found');
+        }
+
+        // Try and initialize command
+        $command = new $commandClass;
+
+        $command->loadOptions();
+
+        if ($command->option('-h', false)) {
+            return $command->help();
+        }
+
+        // Run command
+        $message = $command->handle();
+
+        // Return output and end
+        $this->complete($message);
+    }
+
+    /**
+     * get command's help information
+     * @return null
+     */
+    public function help()
+    {
+        $help = new Help(new OptionFactory);
+
+        $help->setSummary($this->summary)
+                ->setDescr($this->description)
+                ->setUsage($this->usage)
+                ->setOptions($this->commandOptions);
+
+        $this->output->outln($help->getHelp($this->name));
+
+        exit();
+    }
+
+    /**
+     * fail the command and die
+     * @param  string $message
+     * @return null
+     */
+    public function fail($message = null)
+    {
+        if ($message) {
+            $this->output->errln("<<red>>{$message}<<reset>>");
+        }
+
+        exit(Status::FAILURE);
+    }
+
+    /**
+     * complete the command and die
+     * @param  string $message
+     * @return null
+     */
+    public function complete($message = null)
+    {
+        if ($message) {
+            $this->output->outln("<<green>>{$message}<<reset>>");
+        }
+
+        exit(Status::SUCCESS);
+    }
+
+    /**
+     * write info to the console
+     * @param  string
+     * @return null
+     */
+    public function write($message = null)
+    {
+        $this->output->outln($message);
+    }
+
+    /**
+     * write info to the console
+     * @param  string
+     * @return null
+     */
+    public function info($message = null)
+    {
+        $this->output->outln("<<green>>{$message}<<reset>>");
+    }
+
+    /**
+     * write error info to the console
+     * @param  string
+     * @return null
+     */
+    public function error($message = null)
+    {
+        $this->output->outln("<<red>>{$message}<<reset>>");
+    }
+
+    /**
+     * Ask question and get input
+     * @param  string $question
+     * @return mixed
+     */
+    public function ask($question, $default = '')
+    {
+        $this->output->out($question . ' ');
+
+        $result = $this->input->in();
+
+        return $result ?? $default;
```

Comment:
Null-coalescing operator needs to be refactored to PHP 5.6 compatible.

Reply:
Fixed in latest commit.